app/views/text_components/index.html.haml

```diff
@@ -38,3 +38,174 @@
+                            :class => "class",
+                            'data-slider-id' => "slider-range",
+                            'data-provide'=> "slider",
+                            'data-slider-min'=>"0",
+                            'data-slider-max'=>"100",
+                            'data-slider-step'=>"1",
+                            'data-slider-range'=>true,
+                            'data-slider-value'=>"[0,100]",
+                            'data-slider-tooltip'=>"always",
+                            'data-slider-tooltip-split'=>"true",
+                            }
+                  .col-md-6
+                    .slider-group
+                      %label Timeliness Constraint
+                      %input{:type => "text",
+                             :name => "somename",
+                             'data-slider-id' => 'slider-single-time',
+                             'data-provide' => "slider",
+                             'data-slider-min'=>"0",
+                             'data-slider-max'=>"10",
+                             'data-slider-step'=>"1",
+                             'data-slider-tooltip'=>"always",
+                             }
+                .row
+                  .col-md-6
+                    .form-group
+                      %label{:for => "Event"} Event
+                      %select.selectpicker
+                        %option 5 Events
+                        %option 2 Events
+                        %option 1 Event
+                  .col-md-6
+                    .slider-group
+                      %label Uhrzeit
+                      %input{:type => "text",
+                            :name => "somename",
+                            'data-slider-id' => "slider-time-range",
+                            'data-provide'=> "slider",
+                            'data-slider-min'=>"0",
+                            'data-slider-max'=>"2400",
+                            'data-slider-step'=>"60",
+                            'data-slider-range'=>true,
+                            'data-slider-value'=>"[0,2400]",
+                            'data-slider-tooltip'=>"show",
+                            'data-slider-tooltip-split'=>"true",
+                            }
+                .row
+                  .col-md-6
+                    .form-group
+                      %label{:for => "Event"} By chance
+                      %label.checkbox-inline
+                        %input#inlineCheckbox1{:type => "checkbox", :value => "option1"}/
+                        %span.checkbox-button yes
+                      %label.checkbox-inline
+                        %input#inlineCheckbox2{:type => "checkbox", :value => "option2"}/
+                        %span.checkbox-button no
+                  .col-md-6
+                    .form-group
+                      %label Triggercolor
+
+                .row
+                  .col-md-12
+                    = render 'form'
```

Comment:
@jtackmann looks good.
Forms are shared across views in Rails. Look at my last commit. I reused `render 'form'`, so the same HTML will be generated in the modal that is generated in the `/text_components/new` and `/text_components/edit` view. For new views, like your modal, you just have to add instance variables like `@text_component = ...` in the controller. You don't have to do that for existing views (makes it easier).